src/Features/Context/GsspContext.php

```diff
@@ -17,14 +17,478 @@
+        $signatureVerifier = new SignatureVerifier($keyLoader, $logger);
+        $redirectBinding = new RedirectBinding($logger, $signatureVerifier, $serviceProviders);
+        $configuration = new ConfigurationContainer([
+            'registration_route' => 'registration_route_action',
+        ]);
+        $stateHandler = new MemoryStateHandler();
+        $this->authenticationRegistrationService = new AuthenticationRegistrationService($stateHandler);
+        $signingService = new AssertionSigningService($this->identityProvider);
+        $this->responseContext = new ResponseContext(
+            $this->identityProvider,
+            $serviceProviders,
+            $stateHandler
+        );
+        $responseService = new ResponseService(
+            $this->identityProvider,
+            $this->responseContext,
+            $signingService
+        );
+        $responseBuilderFactory = new ResponseBuilderFactory();
+        $this->controller = new IdentityController(
+            $redirectBinding,
+            $configuration,
+            $stateHandler,
+            $responseService,
+            $this->responseContext,
+            $responseBuilderFactory
+        );
+        $container = \Mockery::mock(ContainerInterface::class);
+        $container->shouldReceive('has')->with('templating')->andReturn(false);
+        $container->shouldReceive('has')->with('twig')->andReturn(true);
+        $this->twigEnvironment = \Mockery::mock(Twig_Environment::class);
+        $this->twigEnvironment->shouldReceive('render')->andReturnUsing(function (
+            $template,
+            $parameters = [],
+            /** The response will be assigned by the step definitions */
+            $response = null
+        ) {
+            $this->twigParameters = $parameters;
+            $this->twigTemplate = $template;
+            return $template.'-response';
+        });
+        $container->shouldReceive('get')->with('twig')->andReturn($this->twigEnvironment);
+
+        $router = \Mockery::mock(RouterInterface::class);
+        $router->shouldReceive('generate')
+            ->with('registration_route_action', [], UrlGeneratorInterface::ABSOLUTE_PATH)
+            ->andReturn('https://identity_provider/registration');
+        $container->shouldReceive('get')->with('router')->andReturn($router);
+        $this->controller->setContainer($container);
+    }
+
+    /**
+     * Creates a normal AuthnRequest and set it to this context.
+     *
+     * @Given a normal SAML 2.0 AuthnRequest
+     *
+     * @throws \Exception
+     */
+    public function createANormalAuthnRequest()
+    {
+        $request = new SAML2_AuthnRequest();
+        $request->setAssertionConsumerServiceURL($this->serviceProvider->getAssertionConsumerUrl());
+        $request->setDestination($this->identityProvider->getSsoUrl());
+        $request->setIssuer($this->serviceProvider->getEntityId());
+        $request->setProtocolBinding(SAML2_Const::BINDING_HTTP_POST);
```

Comment:
Using the `SAML2_Const::BINDING_HTTP_REDIRECT` protocol binding is more in line with how this service will be used.